docs/features/search/declarative-integration.md

```diff
@@ -0,0 +1,229 @@
+# Declarative Integrated Search Plugin
+
+> **Disclaimer:**
+> Declarative integration is in an experimental stage and is not recommended for production.
+
+This is a guide for experimenting with `Search` in a declarative integrated Backstage front-end application.
+
+## Main Concepts
+
+Using declarative integration, you can customize your Backstage instance without writing code, see this [RFC](https://github.com/backstage/backstage/issues/18372) for more information.
+
+In this new system architecture, everything that extends the core Backstage features is called a extension, so a extension could be since an api to a page component, which means all plugins can provide are extensions.
```

Comment:
This doesn't read right

Reply:
Thanks Tika, just rephrased the sentence.